lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -0,0 +1,1236 @@
+    promote:
+        description: Set to True to promote a read replica cluster.
+        type: bool
+        default: False
+    purge_cloudwatch_logs_exports:
+        description: Set to False to retain any enabled cloudwatch logs that aren't specified in the task and are associated with the instance.
+        type: bool
+        default: True
+    purge_tags:
+        description: Set to False to retain any tags that aren't specified in task and are associated with the instance.
+        type: bool
+        default: True
+    wait:
+        description: Whether to wait for the cluster to be available or deleted.
+        type: bool
+        default: True
+
+    # Options that have a corresponding boto3 parameter
+    allocated_storage:
+        description:
+          - The amount of storage (in gibibytes) to allocate for the DB instance.
+    allow_major_version_upgrade:
+        description:
+          - Whether to allow major version upgrades.
+        type: bool
+    apply_immediately:
+        description:
+          - A value that specifies whether modifying a cluster with I(new_db_instance_identifier) and I(master_user_password)
+            should be applied as soon as possible, regardless of the I(preferred_maintenance_window) setting. If false, changes
+            are applied during the next maintenance window.
+        type: bool
+        default: False
+    auto_minor_version_upgrade:
+        description:
+          - Whether minor version upgrades are applied automatically to the DB instance during the maintenance window.
+        type: bool
+    availability_zones:
+        description:
+          - A list of EC2 Availability Zones that instances in the DB cluster can be created in.
+            May be used when creating a cluster or when restoring from S3 or a snapshot.
+        aliases:
+          - zones
+          - az
+    backup_retention_period:
+        description:
+          - The number of days for which automated backups are retained (must be greater or equal to 1).
+            May be used when creating a new cluster, when restoring from S3, or when modifying a cluster.
+    ca_certificate_identifier:
+        description:
+          - The identifier of the CA certificate for the DB instance.
+    character_set_name:
+        description:
+          - The character set to associate with the DB cluster.
+    copy_tags_to_snapshot:
+        description:
+          - Whether or not to copy all tags from the DB instance to snapshots of the instance. When initially creating
+            a DB instance the RDS API defaults this to false if unspecified.
+        type: bool
+    db_cluster_identifier:
+        description:
+          - The DB cluster (lowercase) identifier. The identifier must contain from 1 to 63 letters, numbers, or
+            hyphens and the first character must be a letter and may not end in a hyphen or contain consecutive hyphens.
+        aliases:
+          - cluster_id
+    db_instance_class:
```

Comment:
I haven't gone through all the options yet but I think it would be great if we could make ``ec2_instance`` and ``rds_instance`` as close as possible in terms of how they are used.
This might translate to changing something here or changing something in ``ec2_instance`` if we feel the implementation here is better.

For example, ``ec2_instance`` has ``instance_type`` for picking the flavor of the instance but here it is ``db_instance_class`` instead.

Reply:
Thanks so much for reviewing @dmsimard. Let me know if you notice anything else.

My implementation relies on the module options being convertible to the CamelCase boto3 method parameters, so I've been adding aliases rather than modifying the options. Does that work for you? I'll comb through the options for any in common in ec2_instance and add them as aliases. I think modifying the options themselves would make the module harder to maintain (note that this module has 75 options whereas ec2_instance has 35; I want to keep things as simple as possible while being user-friendly).